Sysdig Secure Source (c2c Collector) (#5037)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/sysdig-secure-source.md

---
id: sysdig-secure-source
title: Sysdig Secure Source
sidebar_label: Sysdig Secure
tags:
  - cloud-to-cloud
  - sysdig-secure

> description: The Sysdig Secure Source aims to collect the complete scan results produced by the scanner from Sysdig API and send them to Sumologic.

Reviewer: ```suggestion
description: The Sysdig Secure Source aims to collect the scan results from the scanner using Sysdig API and send them to Sumologic.
```

---

import CodeBlock from '@theme/CodeBlock';
import ExampleJSON from '/files/c2c/sysdig-secure/example.json';
import MyComponentSource from '!!raw-loader!/files/c2c/sysdig-secure/example.json';
import TerraformExample from '!!raw-loader!/files/c2c/sysdig-secure/example.tf';
import ForwardToSiem from '/docs/reuse/forward-to-siem.md';
import useBaseUrl from '@docusaurus/useBaseUrl';

<img src={useBaseUrl('img/integrations/misc/sysdig-logo.png')} alt="icon" width="125"/>

> Sysdig Secure is a security platform that helps monitor and protect cloud-native environments, especially those using mordern containers and Kubernetes. It helps organizations detect, prevent, and respond to security threats while ensuring compliance in real-time.

Reviewer: ```suggestion
Sysdig Secure is a comprehensive security platform that provides continuous security and compliance monitoring for cloud-native environments. This is designed specifically to address the security needs of modern containerised and Kubernetes infrastructures. Sysdig Secure enables organisations to detect, prevent, and respond to security threats and ensure compliance in real-time.
```

## Data collected

| Polling Interval | Data | Description |
| :--- | :--- | :--- |

> | 24 hours | Runtime Scan Result | Provides a list of runtime scan results discovered by the runtime scanner. |

Reviewer: Don't we have the API doc page path for the data?

Author: @priyansh-sumo can you please provide links if the API docs are public?

Author: added the links

| 5 minutes | Full Scan Result | Provides the full scan details of the provided scan result ID. |

## Setup

### Vendor configuration

The Sysdig Secure Source supports API token-based authentication and requires you to provide the Sysdig Secure Base URL and API token to access the data.

#### Base URL

> The **Base URL** is the Sysdig Secure domain URL that corresponds to the region in which the account is created. For example, `https:<hostname>`.

Reviewer: ```suggestion
The **Base URL** is the Sysdig Secure domain URL that corresponds to the region in which the account is created. For example, `https://api.us2.sysdig.com`.
```

#### Bearer token

> The **Bearer Token** is the API security token used to authenticate with Sysdig HTTP API.

Reviewer: ```suggestion
The **Bearer Token** is the API security token used to authenticate with Sysdig HTTP API. Follow the steps below to retrieve the Sysdig API token:
```

> Follow the steps below to retrieve the Sysdig API token:

Reviewer: ```suggestion
```

> 1. Log in to **Sysdig Secure** and then select **Settings** from the user menu.

Reviewer: ```suggestion
1. Sign in to **Sysdig Secure** platform and then select **Settings** from the user menu.
```

> 1. Click **User Profile** to view the **Sysdig Secure API** token in the UI.

Reviewer: ```suggestion
1. Select **User Profile** to view the Sysdig Secure API token in the UI.
```

1. On the Collection page, click **Add Source** next to a Hosted Collector.
1. Search for and select **Sysdig Secure**.
1. Enter a **Name** for the Source. The description is optional.
1. (Optional) For **Source Category**, enter any string to tag the output collected from the Source. Category metadata is stored in a searchable field called `_sourceCategory`.
1. (Optional) **Fields**. Click the **+Add** button to define the fields you want to associate. Each field needs a name (key) and value.
   * ![green check circle.png](/img/reuse/green-check-circle.png) A green circle with a check mark is shown when the field exists in the Fields table schema.
   * ![orange exclamation point.png](/img/reuse/orange-exclamation-point.png) An orange triangle with an exclamation point is shown when the field doesn't exist in the Fields table schema. In this case, an option to automatically add the nonexistent fields to the Fields table schema is provided. If a field is sent to Sumo Logic that does not exist in the Fields schema is ignored, known as dropped.

> 1. **Base URL**. Enter the [Sysdig Secure platform](#base-url) Base URL. For example, `https://eu1.app.sysdig.com`.

Reviewer: ```suggestion
1. **Base URL**. Enter the [Sysdig Secure platform](#base-url) Base URL. For example, `https://api.us2.sysdig.com`.
```

> 1. **Bearer Token**. Enter the Sysdig Secure API token collected from the [Sysdig Secure platform](#bearer-token). For example, `t3fPdsbxxxxxxxxxp4D6hbi4`.

Reviewer: ```suggestion
1. **Bearer Token**. Enter the Sysdig Secure API token collected from the [Sysdig Secure](#bearer-token) platform. For example, `t3fPdsbxxxxxxxxxp4D6hbi4`.
```

> 1. (Optional) **Processing Rules for Logs**. Configure any desired filters, such as allowlist, denylist, hash, or mask, as described in [Create a Processing Rule](/docs/send-data/collection/processing-rules/create-processing-rule).

Reviewer: Enter the information for Filters.

### Configuration Object

| Parameter | Type | Required | Default | Description | Example |
|:--|:--|:--|:--|:--|:--|
| name | String | Yes | `null` | Type a desired name of the source. The name must be unique per Collector. This value is assigned to the [metadata](/docs/search/get-started-with-search/search-basics/built-in-metadata) field `_source`. | `"mySource"` |
| description | String | No | `null` | Type a description of the source. | `"Testing source"` |
| category | String | No | `null` | Type a category of the source. This value is assigned to the [metadata](/docs/search/get-started-with-search/search-basics/built-in-metadata) field `_sourceCategory`. See [best practices](/docs/send-data/best-practices) for details. | `"mySource/test"` |
| fields | JSON Object | No | `null` | JSON map of key-value fields (metadata) to apply to the collector or source. Use the boolean field `_siemForward` to enable forwarding to SIEM.| `{"_siemForward": false, "fieldA": "valueA"}` |
| baseURL | String | Yes | `null` | The Sysdig base URL of your region. | `https://api.us2.sysdig.com` |

> | apiToken | String | Yes | `null` | The API Token of your Sysdig account. | `t3fPdsbxxxxxxxxxp4D6hbi4` |

Reviewer: ```suggestion
| apiToken | String | Yes | `null` | The API token of your Sysdig account. | `t3fPdsbxxxxxxxxxp4D6hbi4` |
```

---
id: sysdig-secure-source
title: Sysdig Secure Source
sidebar_label: Sysdig Secure
tags:
  - cloud-to-cloud
  - sysdig-secure

> description: The Sysdig Secure Source aims to collect the scan results from the scanner using Sysdig API and send them to Sumologic.

Reviewer: ```suggestion
description: The Sysdig Secure Source aims to collect the scan results from the scanner using Sysdig API and send them to Sumo Logic.
```

Sysdig Secure is a comprehensive security platform that provides continuous security and compliance monitoring for cloud-native environments. This is designed specifically to address the security needs of modern containerised and Kubernetes infrastructures. Sysdig Secure enables organisations to detect, prevent, and respond to security threats and ensure compliance in real-time.

## Data collected

| Polling Interval | Data |
| :--- | :--- |
| 24 hours | [Runtime Scan Result](https://us2.app.sysdig.com/apidocs/secure/#tag/Vulnerability-Management/operation/ScannerAPIServiceListRuntimeResults) |
| 5 minutes | [Full Scan Result](https://us2.app.sysdig.com/apidocs/secure/#tag/Vulnerability-Management/operation/GetSecureVulnerabilityV1ResultsResultId) |

## Setup

### Vendor configuration

The Sysdig Secure Source supports API token-based authentication and requires you to provide the Sysdig Secure Base URL and API token to access the data.

#### Base URL

The **Base URL** is the Sysdig Secure domain URL that corresponds to the region in which the account is created. For example, `https://api.us2.sysdig.com`.

#### Bearer token

> The **Bearer Token** is the API security token used to authenticate with Sysdig HTTP API. Follow the steps below to retrieve the Sysdig API token:

Reviewer: ```suggestion
The **Bearer Token** is the API security token used to authenticate with Sysdig HTTP API. Follow the steps below to collect your Sysdig API token:
```

> 

Reviewer: ```suggestion
```

> 1. Sign in to **Sysdig Secure** platform .

Reviewer: ```suggestion
1. Sign in to the **Sysdig Secure** platform.
```

1. Select **Settings** from the **User Menu**.

> 1. Select **User Profile** to view the Sysdig Secure API token in the UI.

Reviewer: ```suggestion
1. In the **Settings** page, select **User Profile** to view the Sysdig Secure API token in the UI.
```
